lib/init/grass.py

```diff
@@ -2367,6 +2374,10 @@ def main():
         start_gui(grass_gui)
         kv = read_gisrc(gisrc)
         kv['PID'] = str(shell_process.pid)
+        # Make PID for this process on macOS, needed for enable quitting
+        # GRASS, trac #3009.
+        if MACOSX:
+            kv['PID'] = str(os.getpid())
         write_gisrc(kv, gisrc)
         exit_val = shell_process.wait()
         if exit_val != 0:
```

Comment:
The line `close_gui()` and onward are not reached on macOS when exiting from GUI (no problem for `close_gui()`, but problem especially for `clean_all()`).

Reply:
`close_gui()` is actually important too if there is more than one GUI running.